pkg/cmd/boot/boot.go

```diff
@@ -184,6 +184,13 @@ func (o *BootOptions) Run() error {
 		return err
 	}
 
+	// check if the user has passed in a requirements file to a cluster which has already been provisioned.
+	if o.RequirementsFile != "" {
```

Comment:
why not integrate this into the existing `overrideRequirements` function? That would be the natural place to do it, right?

I am also wondering whether the criteria of an empty `o.RequirementsFile` does not fall a bit short? 

```go
	requirements, requirementsFile, err := config.LoadRequirementsConfig(o.Dir, config.DefaultFailOnValidationError)
	if err != nil {
		return errors.Wrapf(err, "loading requirements from dir %q", o.Dir)
	}

	// overwrite the default requirements with provided requirements
	if o.RequirementsFile != "" {
		providedRequirements, err := config.LoadRequirementsConfigFile(o.RequirementsFile, config.DefaultFailOnValidationError)
		if err != nil {
			return errors.Wrapf(err, "loading requirements from file %q", o.RequirementsFile)
		}
		*requirements = *providedRequirements
	}
```

This is the code from `overrideRequirements`. We already try to find the requirements file in the current directory. I would do an `filepath.ABS` on both path (provided both are set) and if they are not identical generate the warning/error.  Probably needs to be paired with the `o.GetDevEnv()` check you added. But it would allow for the cases where the user does something like `jx boot -r ./jx-requirements.yml` or from a sub-directory `jx boot -r ../jx-requirements.yml`. WDYT?

Reply:
Maybe we can write a test for this as well.

Reply:
I guess it comes down to the intention of the `--requirements` option, I've always been of the opinion that it should only be used for the initial boot sequence of a fresh cluster.

Reply:
> This is the code from `overrideRequirements`. We already try to find the requirements file in the current directory. I would do an `filepath.ABS` on both path (provided both are set) and if they are not identical generate the warning/error. Probably needs to be paired with the `o.GetDevEnv()` check you added. But it would allow for the cases where the user does something like `jx boot -r ./jx-requirements.yml` or from a sub-directory `jx boot -r ../jx-requirements.yml`. WDYT?

I think it's dangerous in general to want to specify a requirements file via the `--requirements` option when `boot`ing an already provisioned cluster as those changes aren't guaranteed to make their way into the environment dev repository unless I've misunderstood something?

Reply:
Ok, so I am coming around here. Once there is an existing Jenkins X install, you should not be able to use the `-r` option.

Reply:
We probably should also add something to the help message:

`cmd.Flags().StringVarP(&options.RequirementsFile, "requirements", "r", "", "requirements file which will overwrite the default requirements file")`